src/cdp-proxy/reactNativeCDPProxy.ts

```diff
@@ -80,7 +80,7 @@ export class ReactNativeCDPProxy {
         this.applicationTarget.onReply(this.handleApplicationTargetReply.bind(this));
         this.debuggerTarget.onReply(this.handleDebuggerTargetReply.bind(this));
 
-        this.applicationTarget.onEnd(this.onApplicationTargetClosed.bind(this));
+        this.debuggerTarget.onEnd(this.onDebuggerTargetClosed.bind(this));
```

Comment:
Why we rename this?

Reply:
I guess that it is more reliable to consider debug session is closed when debugger target has been closed. `applicationTarget.onEnd` method can be called even in case the current debug session hasn't been already terminated. So I changed this behavior. This means that we restart debugging only in case debug session has been closed.